pkg/await/apps_deployment.go

```diff
@@ -503,6 +541,31 @@ func (dia *deploymentInitAwaiter) changeTriggeredRollout() bool {
 	return len(fields) > 0
 }
 
+func (dia *deploymentInitAwaiter) checkPersistentVolumeClaimStatus() {
+	inputs := dia.config.currentInputs
+
+	glog.V(3).Infof("Checking PersistentVolumeClaims status for Deployment '%s'", inputs.GetName())
+
+	allPVCsReady := true // bool used to do a rolling AND operation across all PVCs statuses
+	for _, pvc := range dia.pvcs {
+		phase, hasConditions := openapi.Pluck(pvc.Object, "status", "phase")
+		if !hasConditions {
+			return
+		}
+
+		// Success occurs when all PersistentVolumeClaims are marked as Bound
+		if phase == statusBound {
+			allPVCsReady = allPVCsReady && true
+		} else {
+			allPVCsReady = allPVCsReady && false
```

Comment:
Not sure I understand this. This appears to be setting `allPVCsReady` to true only in the case where it was false to start. Rewriting the logic a bit produces:
```go
if phase != statusBound {
    allPVCsReady = !allPVCsReady
    ...
}
```

My guess is that you meant something like

```go
if phase != statusBound {
    allPVCsReady = false
    // Log errors ...
}
```

Reply:
Ya, I'm not sure what I was thinking. I heeded your advice and simplified the logic - it should be straight forward now.